File: PRACTICA PRIMER PARCIAL/Ejercicio14/Python/GenerarSecuencia.py
def generar_secuencia(n):
    secuencia = []
    indice = 1
    incremento = 1

    for i in range(n):
        if i % 3 == 2:
            secuencia.append(-5)
        else:
            secuencia.append(indice)
            indice += incremento
            incremento += 1

    return secuencia

n_terminos = 20
secuencia = generar_secuencia(n_terminos)
print(secuencia)

File: PRACTICA PRIMER PARCIAL/Ejercicio14/Java/GenerarSecuencia.java
import java.util.ArrayList;
import java.util.List;

public class GenerarSecuencia {

    public static List<Integer> generarSecuencia(int n) {
        List<Integer> secuencia = new ArrayList<>();
        int indice = 1;
        int incremento = 1;

        for (int i = 0; i < n; i++) {
            if (i % 3 == 2) {
                secuencia.add(-5);
            } else {
                secuencia.add(indice);
                indice += incremento;
                incremento++;
            }
        }

        return secuencia;
    }

    public static void main(String[] args) {
        int nTerminos = 20;
        List<Integer> secuencia = generarSecuencia(nTerminos);

        for (int num : secuencia) {
            System.out.print(num + " ");
        }
    }
}
